jcl/src/java.base/share/classes/openj9/internal/tools/attach/target/DiagnosticUtils.java

```diff
@@ -579,20 +595,76 @@ private static DiagnosticProperties doHelp(String diagnosticCommand) {
 
 	}
 
-/*[IF CRAC_SUPPORT]*/
-	private static DiagnosticProperties doCheckpointJVM(String diagnosticCommand) {
-		Thread checkpointThread = new Thread(() -> {
-			try {
-				jdk.crac.Core.checkpointRestore();
-			} catch (Throwable t) {
-				t.printStackTrace();
+/* [IF CRAC_SUPPORT | CRIU_SUPPORT] */
+	private static DiagnosticProperties parseCheckpointCommands(String diagnosticCommand) {
+		DiagnosticProperties result = null;
+		String[] parts = diagnosticCommand.split(DIAGNOSTICS_OPTION_SEPARATOR);
+		// parts[0] is DIAGNOSTICS_CRIU_CHECKPOINT or DIAGNOSTICS_JDK_CHECKPOINT
+		if (parts.length > 1) {
+			// the command format: [CRIU|JDK].checkpoint,imageDir=/path/to/cpData,logLevel=4
```

Comment:
The additional JCMD options should only apply to `CRIU.checkpoint` not `JDK.checkpoint`. The `JDK.checkpoint` does not take any parameters.

Reply:
Updated to disallow any parameters for `JDK.checkpoint`.

@tajila this is ready for another look.